tests/framework/test_time.py

```diff
@@ -246,43 +283,106 @@ def test_uneven_steps(SimulationContext, base_config):
-            ]
-        ).sort_index(),
-    }
 
     ## Ensure that steps and active simulants are correct through one cycle of 21
     for correct_step_size, group in zip(correct_step_sizes, groups):
         validate_step_column_is_pipeline(sim)
         take_step_and_validate(
             sim,
             listener,
-            expected_simulants=test[group],
+            expected_simulants=get_pop_by_parity(sim, group).index,
             expected_step_size_days=correct_step_size,
         )
 
         sample_pipeline = step_modifier_component.ts_pipeline_value
-        assert sample_pipeline.index.equals(test[group])
-        assert np.all(sample_pipeline == pipeline_values[group])
+        assert sample_pipeline.index.equals(get_pop_by_parity(sim, group).index)
+        assert np.all(sample_pipeline == pipeline_by_parity(sim, step_modifiers, group))
+
+
+def test_partial_modification(SimulationContext, base_config):
+    """Test that if we have one modifier that doesn't apply to all simulants,
+    we choose the default value for unmodified simulants.
+    """
+
+    base_config["time"]["step_size"] = 1
+    listener = Listener("listener")
+    ## Define odds for validation, but don't pass it into the step modifier
+    step_modifiers = {"evens": 3, "odds": 1}
+    step_modifier_component = StepModifier(
+        "step_modifier", step_modifiers["evens"], modified_simulants="evens"
+    )
+    sim = SimulationContext(
+        base_config,
+        [step_modifier_component, listener],
+    )
+
+    sim.setup()
+    sim.initialize_simulants()
+    ## We should update odds with default each time, and update evens every third step.
+    correct_step_sizes = [1, 1, 1, 1, 1, 1, 1, 1, 1]
+    groups = ["odds", "odds", "all", "odds", "odds", "all", "odds", "odds", "all"]
+
+    ## Ensure that steps and active simulants are correct through one cycle of 21
+    for correct_step_size, group in zip(correct_step_sizes, groups):
+        validate_step_column_is_pipeline(sim)
+        take_step_and_validate(
+            sim,
+            listener,
+            expected_simulants=get_pop_by_parity(sim, group).index,
+            expected_step_size_days=correct_step_size,
+        )
+
+        sample_pipeline = step_modifier_component.ts_pipeline_value
+        assert sample_pipeline.index.equals(get_pop_by_parity(sim, group).index)
+        assert np.all(sample_pipeline == pipeline_by_parity(sim, step_modifiers, group))
+
+
+def test_multiple_modifiers(SimulationContext, base_config):
+    """Test that if we have a mix of step sizes, we take steps in accordance
+    to reach all simulants' next event times in the fewest steps.
+    """
+
+    base_config["time"]["step_size"] = 1
+    listener = Listener("listener")
+    step_modifiers = {"evens": 3, "odds": 7}
+    step_modifier_A = StepModifier(
+        "step_modifier_A", step_modifiers["evens"], modified_simulants="evens"
+    )
+    step_modifier_B = StepModifier(
```

Comment:
These two modifiers are going to modify disjoint sets of simulants, so we have no simulants who have a modifier from both of them. Is that what we want? Shouldn't we test that we get the expected (minimum) value if two different modifiers modify the same simulant?

Reply:
This does get tested in the test of the step size post-processor, though one might argue that we don't quite test the full integration. My concern in including this and the partial modification test is just to ensure that we are doing proper handling of NaNs when we have modifiers that don't return the full population index.